src/test-c-gen/c_test_source_factory.cpp

```diff
@@ -0,0 +1,138 @@
+#include "c_test_source_factory.h"
+
+#include <util/irep.h>
+#include <util/symbol_table.h>
+
+#include <sstream>
+
+#include <test-c-gen/expr2cleanc.h>
+
+#include <test-c-gen/c_test_file.h>
+#include <test-c-gen/function_parameter_builder.h>
+#include <test-c-gen/function_return_builder.h>
+
+
+/*std::string generate_c_test_case_from_inputs(
+    const symbol_tablet &st, const irep_idt &func_id,
+    bool enters_main, inputst inputs,
+    const interpretert::list_input_varst& opaque_function_returns,
+    const interpretert::input_var_functionst& input_defn_functions,
+    const interpretert::dynamic_typest& dynamic_types,
+    const std::string &test_func_name,
+    const interpretert::side_effects_differencet &valuesDifference,
+    const exprt &assertExpr, bool emitAssert,
+    bool disable_mocks,
+    bool disable_verify_mocks,
+    const optionst::value_listt& mock_classes,
+    const optionst::value_listt& no_mock_classes,
+    const std::vector<std::string>& goals_reached,
+    const std::string& expect_exception)
+{
+  return "int main(int argc, char* argv) { return 0; }";
+}*/
+
+namespace
+{
+/*******************************************************************\
+ Function: filter_inputs_to_function_parameters
+ Inputs:
+  all_inputs - All the inputs that have come back from the interpreter
+  func_expr - The function call to filter for
+Outputs: The inputs which are parameters for the supplied function
+Purpose: To pull out from the inputs the relevant ones for the function
+         call
+\*******************************************************************/
+  inputst filter_inputs_to_function_parameters(const inputst &all_inputs,
```

Comment:
I took the structure from the Java version before I really understood. I've removed the factory file (it just had a bunch of functions relating to generation that were called from the generator so distinction made no sense). 

At the moment my main issue is that we pass a lot of parameters around. Would it be preferable to move the parameters into a constructor and then have a simple call to generate the tests. I've seen this style elsewhere.